atomix/cluster/src/test/java/io/atomix/raft/roles/PassiveRoleTest.java

```diff
@@ -151,4 +152,44 @@ public void shouldNotFlushIfNoEntryIsAppended() {
     verify(log, never()).flush();
     assertThat(response.lastLogIndex()).isEqualTo(0);
   }
+
+  @Test
+  public void shouldStoreLastFlushedIndex() {
+    // given
+    final List<PersistedRaftRecord> entries =
+        List.of(new PersistedRaftRecord(1, 1, 1, 1, new byte[1]));
+    final AppendRequest request = new AppendRequest(2, "", 0, 0, entries, 1);
+
+    when(log.append(any(PersistedRaftRecord.class))).thenReturn(mock(IndexedRaftLogEntry.class));
+    when(ctx.getLog()).thenReturn(log);
+
+    // when
+    role.handleAppend(request).join();
+
+    // then
+    verify(ctx).setFlushedIndex(eq(1L));
+  }
+
+  @Test
+  public void shouldStoreLastFlushedWithSomeFailure() {
```

Comment:
Since this test needs more control that what we can get via high level tests, this is a good candidate for test using mocks. So we can keep this one.